filebeat/prospector/redis/harvester.go

```diff
@@ -0,0 +1,131 @@
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/util"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+
+	"github.com/elastic/beats/filebeat/harvester"
+	rd "github.com/garyburd/redigo/redis"
+	"github.com/satori/go.uuid"
+)
+
+type Harvester struct {
+	outlet    channel.Outleter
+	id        uuid.UUID
+	done      chan struct{}
+	conn      rd.Conn
+	forwarder harvester.Forwarder
+}
+
+// log contains all data related to one slowlog entry
+//
+// 	The data is in the following format:
+// 	1) (integer) 13
+// 	2) (integer) 1309448128
+// 	3) (integer) 30
+// 	4) 1) "slowlog"
+// 	   2) "get"
+// 	   3) "100"
+//
+type log struct {
+	id        int64
+	timestamp int64
+	duration  int
+	cmd       string
+	key       string
+	args      []string
+}
+
+func NewHarvester(outlet channel.Outleter, conn rd.Conn) *Harvester {
+	return &Harvester{
+		id:     uuid.NewV4(),
+		done:   make(chan struct{}),
+		outlet: outlet, // All harvester can use the same outlet as not state
+		conn:   conn,
+	}
+}
+
+func (h *Harvester) Start() error {
+	defer h.conn.Close()
+
+	logp.Debug("redis", "Start slowlog fetch.")
+
+	// Writes Slowlog get and slowlog reset both to the buffer so they are executed together
+	h.conn.Send("SLOWLOG", "GET")
+	h.conn.Send("SLOWLOG", "RESET")
+
+	// Flush the buffer to execute both commands and receive the reply from SLOWLOG GET
+	h.conn.Flush()
+
+	// Receives first reply from redis which is the one from GET
+	logs, err := rd.Values(h.conn.Receive())
+	if err != nil {
+		return fmt.Errorf("error receiving slowlog data: %s", err)
+	}
+
+	// Read reply from RESET
+	_, err = h.conn.Receive()
+	if err != nil {
+		return fmt.Errorf("error receiving reset data: %s", err)
+	}
+
+	for _, item := range logs {
+		entry, err := rd.Values(item, nil)
+		if err != nil {
+			logp.Err("Error loading slowlog values: %s", err)
+			continue
+		}
+
+		var log log
+		var args []string
+		rd.Scan(entry, &log.id, &log.timestamp, &log.duration, &args)
+
+		// This splits up the args into cmd, key, args. Should there also be an entry with
+		// all in one line.
+		argsLen := len(args)
+		if argsLen >= 1 {
+			log.cmd = args[0]
+		}
+		if argsLen >= 2 {
+			log.key = args[1]
+		}
+		// TODO: Could args contain confidential data? Should it be optional?
+		if argsLen >= 3 {
+			log.args = args[2:]
+		}
+
+		data := util.NewData()
+		data.Event = common.MapStr{
+			// TODO: Should we still send the "fetching" timestamp or overwrite it by timestamp?
+			"@timestamp": common.Time(time.Now()),
+			"redis": common.MapStr{
+				"slowlog": common.MapStr{
+					"timestamp": common.Time(time.Unix(log.timestamp, 0).UTC()),
+					"id":        log.id,
+					"cmd":       log.cmd,
+					"key":       log.key,
+					"args":      log.args,
+					"duration": common.MapStr{
+						"us": log.duration,
+					},
+				},
+			},
+		}
+
+		h.forwarder.Forward(data)
+	}
+	return nil
+}
+
+func (h *Harvester) Stop() {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method Harvester.Stop should have comment or be unexported